test_filter.py

```diff
@@ -0,0 +1,151 @@
+        adapter = casbin.persist.adapters.FilteredAdapter("examples/rbac_with_domains_policy.csv")
+        e = casbin.Enforcer("examples/rbac_with_domains_model.conf", adapter)
+        filter = Filter()
+        filter.P = ["", "domain1"]
+        filter.G = ["", "", "domain1"]
+        try:
+            e.load_policy()
+        except:
+            raise RuntimeError("nexpected error in LoadFilteredPolicy")
+        self.assertTrue(e.has_policy(['admin', 'domain1', 'data1','read']))
+        self.assertTrue(e.has_policy(['admin', 'domain2', 'data2','read']))
+        try:
+            e.load_filtered_policy(filter)
+        except:
+            raise RuntimeError("unexpected error in LoadFilteredPolicy")
+
+        if not e.is_filtered:
+            raise RuntimeError("adapter did not set the filtered flag correctly")
+
+        self.assertTrue(e.has_policy(['admin', 'domain1', 'data1','read']))
+        self.assertFalse(e.has_policy(['admin', 'domain2', 'data2','read']))
+
+        filter.P = ["", "domain2"]
+        filter.G = ["", "", "domain2"]
+        try:
+            e.load_increment_filtered_policy(filter)
+        except:
+            raise RuntimeError("unexpected error in LoadFilteredPolicy")
+
+        self.assertTrue(e.has_policy(['admin', 'domain1', 'data1','read']))
+        self.assertTrue(e.has_policy(['admin', 'domain2', 'data2','read']))
+
+
+    def test_filtered_policy_invalid_filter(self):
+        adapter = casbin.persist.adapters.FilteredAdapter("examples/rbac_with_domains_policy.csv")
+        e = casbin.Enforcer("examples/rbac_with_domains_model.conf", adapter)
+        filter = ["","domain1"]
+        try:
+            e.load_filtered_policy(filter)
+        except:
+            pass
+        else:
+            raise RuntimeError("expected error in LoadFilteredPolicy, but got no error")
+
+
+    def test_filtered_policy_empty_filter(self):
+        adapter = casbin.persist.adapters.FilteredAdapter("examples/rbac_with_domains_policy.csv")
+        e = casbin.Enforcer("examples/rbac_with_domains_model.conf", adapter)
+
+        try:
+            e.load_filtered_policy(None)
+        except:
+            raise RuntimeError("unexpected error in LoadFilteredPolicy")
+
+        if e.is_filtered():
+            raise RuntimeError("adapter did not reset the filtered flag correctly")
+
+        try:
+            e.save_policy()
+        except:
+            raise RuntimeError("unexpected error in SavePolicy")
+
+
+    def test_unsupported_filtered_policy(self):
+        e = casbin.Enforcer("examples/rbac_with_domains_model.conf", "examples/rbac_with_domains_policy.csv")
+        filter = Filter()
+        filter.P = ["", "domain1"]
+        filter.G = ["", "", "domain1"]
+        try:
+            e.load_filtered_policy(filter)
+        except : pass
+        else:
+            raise RuntimeError("encorcer should have reported incompatibility error")
+
+    def test_filtered_adapter_empty_filepath(self):
```

Comment:
In [Go version](https://github.com/casbin/casbin/blob/0692af5775a3e8100e569256999243b9b6299c9b/filter_test.go#L148)
if err != nil,raising a error
but in python version,I just pass.The func e.load_filtered_policy() must raise an error,I just ignore the error to pass the unittest.
I dont't konw if it's right to do that.

Reply:
The exception can be asserted:
```
with self.assertRaises(RuntimeError):
    // do something
```